unitary/alpha/quantum_world.py

```diff
@@ -96,13 +100,29 @@ def combine_with(self, other_world: "QuantumWorld"):
         # Clear the other world so that objects cannot be used from that world.
         other_world.clear()
 
+    def _append_op(self, op: cirq.Operation):
+        """Add the operation in a way designed to speed execution.
+
+        For the sparse simulator post-selections should be as early as possible to cut
+        down the state size. Also X's since they don't increase the size.
+        """
+        if not self.use_sparse:
+            self.circuit.append(op)
+            return
+
+        if isinstance(op, PostSelectOperation) or op.gate is cirq.X:
+            strategy = cirq.InsertStrategy.EARLIEST
```

Comment:
If everyone is EARLIEST, no one is... I add most operations with NEW to have them added at the end, so that they don't get placed ahead of post-selection operations which work better if they are earlier.